Sample_v3/Samples/CombineSimpleChat/CombineSimpleChatViewController.swift

```diff
@@ -0,0 +1,436 @@
+    /// The editing is allowed for non-deleted messages
+    ///
+    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        channelController.messages[indexPath.row].deletedAt == nil
+    }
+    
+    ///
+    /// # willBeginEditingRowAt
+    ///
+    /// In this method we get a swipe container and apply the same transform as the tableView has.
+    /// It fixes the bug when the list is in `listOrdering == .bottomToTop` but the swipe actions are
+    /// upside-down
+    ///
+    override func tableView(_ tableView: UITableView, willBeginEditingRowAt indexPath: IndexPath) {
+        tableView.cellForRow(at: indexPath)?.swipeActionsContainer?.transform = tableView.transform
+    }
+    
+    ///
+    /// # trailingSwipeActionsConfigurationForRowAt
+    ///
+    /// The method below returns a list of swipe actions available on the message
+    ///
+    override func tableView(
+        _ tableView: UITableView,
+        trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath
+    ) -> UISwipeActionsConfiguration? {
+        let message = channelController.messages[indexPath.row]
+        
+        let messageController = chatClient.messageController(
+            cid: channelController.channelQuery.cid,
+            messageId: message.id
+        )
+        
+        let deleteAction = UIContextualAction(style: .destructive, title: "Delete") { _, _, _ in
+            messageController.deleteMessage()
+        }
+        
+        let editAction = UIContextualAction(style: .normal, title: "Edit") { [weak self] _, _, _ in
+            self?.showTextEditingAlert(for: message.text) {
+                messageController.editMessage(text: $0)
+            }
+        }
+        
+        let configuration = UISwipeActionsConfiguration(actions: [deleteAction, editAction])
+        configuration.performsFirstActionWithFullSwipe = true
+        return configuration
+    }
+    
+    // MARK: - Button Actions
+
+    ///
+    /// The methods below are called when the user presses some button in the interface to send a message or open the channel menu.
+    ///
+    
+    ///
+    /// # sendMessageButtonTapped
+    ///
+    /// The method below is called when the user taps the send button. To send the message, `channelController?.createNewMessage(text:)` is called.
+    ///
+    @objc func sendMessageButtonTapped(_ sender: Any) {
+        guard let text = composerView.textView.text else {
+            return
+        }
+
+        channelController?.createNewMessage(text: text)
+        
+        composerView.textView.text = ""
+    }
+    
+    ///
+    /// # showChannelActionsAlert
+    ///
+    /// The method below displays a `UIAlertController` with many actions that can be taken on the `channelController` such as `addMembers`, `removeMembers`,
+    /// and `deleteChannel`.
+    ///
+    @objc func showChannelActionsAlert() {
```

Comment:
Can you copy latest version of this from `SimpleChatViewController.swift` ? It allows specifying member id for the actions.